diesel/src/query_builder/sql_query.rs

```diff
@@ -164,3 +190,81 @@ where
 }
 
 impl<Conn, Query, Value, ST> RunQueryDsl<Conn> for UncheckedBind<Query, Value, ST> {}
+
+#[must_use = "Queries are only executed when calling `load`, `get_result`, or similar."]
+/// See [`SqlQuery::into_boxed`].
+///
+/// [`SqlQuery::into_boxed`]: ../struct.SqlQuery.html#method.into_boxed
+#[allow(missing_debug_implementations)]
+pub struct BoxedSqlQuery<'f, DB: Backend, Query> {
+    query: Query,
+    sql: String,
+    binds: Vec<Box<dyn Fn(AstPass<DB>) -> QueryResult<()> + 'f>>,
+}
+
+impl<'f, DB: Backend, Query> BoxedSqlQuery<'f, DB, Query> {
+    pub(crate) fn new(query: Query) -> Self {
+        BoxedSqlQuery {
+            query,
+            sql: "".to_string(),
+            binds: vec![],
+        }
+    }
+
+    /// See [`SqlQuery::bind`].
+    ///
+    /// [`SqlQuery::bind`]: ../struct.SqlQuery.html#method.bind
+    pub fn bind<BindSt, Value>(mut self, b: Value) -> Self
+    where
+        DB: HasSqlType<BindSt>,
+        Value: ToSql<BindSt, DB> + 'f,
+    {
+        self.binds
+            .push(Box::new(move |mut out| out.push_bind_param_value_only(&b)));
+        self
+    }
+
+    /// See [`SqlQuery::sql`].
+    ///
+    /// [`SqlQuery::sql`]: ../struct.SqlQuery.html#method.sql
+    pub fn sql<T: Into<String>>(mut self, sql: T) -> Self {
+        self.sql += &sql.into();
```

Comment:
It should be possible to avoid calling `Into<String>` here, so that we could save one allocation.
Maybe something like:

```rust
pub fn sql<T: AsRef<str>>(mut self, sql: T) -> Self {
        self.sql += sql.as_ref()
        self
)
```

(The same should be done for the sql function of `SqlQuery`)